### PR TITLE
Update Scala 3 version to 0.23.0

### PR DESCRIPTION
**Pull request by tgodzik**
Let's move to a stable version before the release.

### CODE REVIEW

**Comment by olafurpg** [comment]
MUnit currently only supports one Dotty version at a time but we can probably support the latest two versions if there is demand. The upgrade from 0.23 to 0.24 didn't require any changes in the implementation.

**Comment by gabro** [comment]
I don’t see much value in supporting multiple dotty versions when the release cycle is of 6 weeks.

**Comment by tgodzik** [comment]
> MUnit currently only supports one Dotty version at a time but we can probably support the latest two versions if there is demand. The upgrade from 0.23 to 0.24 didn't require any changes in the implementation.

I think we should support a single Dotty version, but if we do that I think it would be clearer to only support stable releases. It's a bit weird to only support release candidates and not the releases themselves. Otherwise, I would support last stable and last release candidate.

**Comment by olafurpg** [comment]
I think it's more weird to not support the latest release since I expect most Dotty users to be early adopters that want to stay on the latest. Still, I think it's a bit rough to not give any sliding window to complete the upgrade so it might make sense to support the latest stable + candidate. That gives people 6 weeks to upgrade to the latest Dotty

**Comment by tgodzik** [comment]
> I think it's more weird to not support the latest release since I expect most Dotty users to be early adopters that want to stay on the latest. Still, I think it's a bit rough to not give any sliding window to complete the upgrade so it might make sense to support the latest stable + candidate. That gives people 6 weeks to upgrade to the latest Dotty

I updated two include the last 2 Dotty versions, though now we will need the munit release. Is it ok to release now or is there anything we are waiting for?

File: mtags/src/main/scala-0.24/scala/scala/meta/internal/pc/CompilerInterfaces.scala [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import dotty.tools.dotc.reporting.Diagnostic.Error
+import dotty.tools.dotc.interfaces.Diagnostic
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.util.SourcePosition
+import dotty.tools.io.VirtualFile
+import org.eclipse.{lsp4j => l}
+
+// note(@tgodzik) the plan is to be able to move the methods here back to Dotty compiler
+// so that we can provide easier compatibility with multiple Scala 3 versions
+object CompilerInterfaces {
```
Comment (tgodzik) [inline mtags/src/main/scala-0.24/scala/scala/meta/internal/pc/CompilerInterfaces.scala:24]: So this is my first attempt to move some of the methods we use to a separate class. In time we might be able to move those methods into Dotty itself and allow for better compatibility with different Scala versions.

Plus this part was incompatible between 0.23 and 0.24

**Comment by olafurpg** [comment]
Here is a PR adding 0.23.0 + 0.24.0-RC1 support in MUnit https://github.com/scalameta/munit/pull/117

**Comment by tgodzik** [comment]
> Here is a PR adding 0.23.0 + 0.24.0-RC1 support in MUnit [scalameta/munit#117](https://github.com/scalameta/munit/pull/117)

Thanks! I updated the PR with the support for the two versions.

File: build.sbt [hunk 2]
```diff
@@ -302,7 +312,7 @@ val mtagsSettings = List(
       crossSetting(
         scalaVersion.value,
         if211 = List("com.lihaoyi" %% "pprint" % "0.5.4"),
-        if2 = List("com.lihaoyi" %% "pprint" % "0.5.9"),
+        ifLaterThan211 = List("com.lihaoyi" %% "pprint" % "0.5.9"),
```
Comment (ckipp01) [inline build.sbt:315]: 👍

File: mtags/src/main/scala-0.23/scala/scala/meta/internal/pc/CompilerInterfaces.scala [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    val freshContext = driver.currentCtx.fresh.setReporter(parsingReporter)
+    val parser = new Parser(sourceFile)(freshContext)
+    parser.parse()
+    val diags = parsingReporter.allErrors
+    diags.flatMap(diagnostic).asJava
+  }
+
+  private def diagnostic(mc: MessageContainer): Option[l.Diagnostic] =
+    if (!mc.pos.exists)
+      None // diagnostics without positions are not supported: https://github.com/Microsoft/language-server-protocol/issues/249
```
Comment (ckipp01) [inline mtags/src/main/scala-0.23/scala/scala/meta/internal/pc/CompilerInterfaces.scala:43]: Probably not ideal, but it _could_ depend on what the diagnostic is, but if it is relevant to an entire file for example, we could just detect it, give it a position at the beginning of the file and display it.

Comment (tgodzik) [reply mtags/src/main/scala-0.23/scala/scala/meta/internal/pc/CompilerInterfaces.scala:43]: Good idea, this message comes from Dotty though. Might have copied it with the code :O

**Pull request merged**